crates/store/re_chunk_store/src/dataframe.rs

```diff
@@ -622,6 +769,98 @@ impl ChunkStore {
         controls.chain(timelines).chain(components).collect()
     }
 
+    /// Given a [`ControlColumnSelector`], returns the corresponding [`ControlColumnDescriptor`].
+    #[allow(clippy::unused_self)]
+    pub fn resolve_control_selector(
+        &self,
+        selector: &ControlColumnSelector,
+    ) -> ControlColumnDescriptor {
+        if selector.component == RowId::name() {
+            ControlColumnDescriptor {
+                component_name: selector.component,
+                datatype: RowId::arrow_datatype(),
+            }
+        } else {
+            ControlColumnDescriptor {
+                component_name: selector.component,
+                datatype: ArrowDatatype::Null,
+            }
+        }
+    }
+
+    /// Given a [`TimeColumnSelector`], returns the corresponding [`TimeColumnDescriptor`].
+    pub fn resolve_time_selector(&self, selector: &TimeColumnSelector) -> TimeColumnDescriptor {
+        let timelines = self.all_timelines();
+
+        let timeline = timelines
+            .iter()
+            .find(|timeline| timeline.name() == &selector.timeline)
+            .copied()
+            .unwrap_or_else(|| Timeline::new_temporal(selector.timeline));
+
+        TimeColumnDescriptor {
+            timeline,
+            datatype: timeline.datatype(),
+        }
+    }
+
+    /// Given a [`ComponentColumnSelector`], returns the corresponding [`ComponentColumnDescriptor`].
+    ///
+    /// If the component is not found in the store, a default descriptor is returned with a null datatype.
+    pub fn resolve_component_selector(
+        &self,
+        selector: &ComponentColumnSelector,
+    ) -> ComponentColumnDescriptor {
+        let datatype = self
+            .lookup_datatype(&selector.component)
+            .cloned()
+            .unwrap_or_else(|| ArrowDatatype::Null);
+
+        let is_static = self
+            .static_chunk_ids_per_entity
+            .get(&selector.entity_path)
+            .map_or(false, |per_component| {
+                per_component.contains_key(&selector.component)
+            });
+
+        // TODO(#6889): Fill `archetype_name`/`archetype_field_name` (or whatever their
+        // final name ends up being) once we generate tags.
+        ComponentColumnDescriptor {
+            entity_path: selector.entity_path.clone(),
+            archetype_name: None,
+            archetype_field_name: None,
+            component_name: selector.component,
+            store_datatype: ArrowListArray::<i32>::default_datatype(datatype.clone()),
```

Comment:
Shouldn't that depend on `selector.join_encoding`?

Reply:
Nope. The `store_datatype` is always a list. `join_encoding` influences the construction of `.returned_datatype()`